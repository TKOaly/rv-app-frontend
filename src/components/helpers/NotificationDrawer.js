import './styles/NotificationDrawer.scss';
import { Fade } from '../animations/Animations';
import { TransitionGroup } from 'react-transition-group';
import { connect } from 'react-redux';
import ErrorNotification from '../notifications/ErrorNotification';
import PurchaseNotification from '../notifications/PurchaseNotification';
import React from 'react';
import SuccessNotification from '../notifications/SuccessNotification';

class NotificationDrawer extends React.Component {
    render = () => {
        return (
            <div className="notificationDrawer">
                <TransitionGroup>
                    {this.props.notifications &&
                        this.props.notifications.length > 0 &&
                        this.props.notifications.map((notification, id) =>
                            notification.messageType === 'SUCCESS' ? (
                                <Fade key={notification.id}>
                                    <SuccessNotification message={notification.message} shadow />
                                </Fade>
                            ) : (
                                <Fade key={notification.id}>
                                    <ErrorNotification message={notification.message} shadow />
                                </Fade>
                            )
                        )}
                    {this.props.purchases && this.props.purchases.length > 0 && (
                        <Fade>
                            <PurchaseNotification shadow purchases={this.props.purchases} />
                        </Fade>
                    )}
                </TransitionGroup>
            </div>
        );
    };
}

const mapStateToProps = (state) => {
    return {
        notifications: state.notification.notifications,
        purchases: state.notification.purchasedItems
    };
};

export default connect(mapStateToProps)(NotificationDrawer);
