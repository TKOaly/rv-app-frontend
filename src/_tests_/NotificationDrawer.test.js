import { shallow } from 'enzyme';
import NotificationDrawer from '../components/helpers/NotificationDrawer';
import React from 'react';
import renderer from 'react-test-renderer';

describe.only('<NotificationDrawer />', () => {
    it('renders correctly', () => {
        const cmpnt = renderer.create(<NotificationDrawer />).toJSON();
        expect(cmpnt).toMatchSnapshot();
    });
});
