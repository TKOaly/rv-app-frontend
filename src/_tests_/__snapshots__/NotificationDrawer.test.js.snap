// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationDrawer /> renders correctly 1`] = `
<div
  className="notificationDrawer"
>
  <div />
</div>
`;
